test/e2e/testsuites/dynamically_provisioned_pod_failover_tester.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"time"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+)
+
+//  will provision required PV(s), PVC(s) and Pod(s)
+// Pod should successfully be re-scheduled on failover in a cluster with AzDriverNode and AzVolumeAttachment resources
+type PodFailover struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	Pod                    PodDetails
+	Volume                 VolumeDetails
+	PodCheck               *PodExecCheck
+	StorageClassParameters map[string]string
+}
+
+func (t *PodFailover) Run(client clientset.Interface, namespace *v1.Namespace) {
+	tDeployment, cleanup := t.Pod.SetupDeployment(client, namespace, t.CSIDriver)
```

Comment:
`StorageClassParameters` is ignored in this test case. In fact, `SetupDeployment` defaults to the storage parameters [here ](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/cdcb750196a0fb1f4b896c69c72bb8e26a8f20d1/test/e2e/driver/azuredisk_driver.go#L130). [My draft PR](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/pull/89) on our private repo fixes the issue. We might need to add some of those changes here as well.

Reply:
can you make the storage class change in this PR? thanks.